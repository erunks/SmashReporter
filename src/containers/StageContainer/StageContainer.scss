@import '../../styles/base.scss';

.StageContainer {
  background-image: conic-gradient(
    from -90deg,
    transparent 0.45turn,
    orange 0.5turn,
    orange 0.55turn,
    $royal_purple 0.85turn,
    $glaicer_freeze 0.95turn,
  );
  background-position: 0% 100%;
  background-size: 100% 200%;
  display: grid;
  grid-template-columns: 40% auto;
  height: 100%;
  position: relative;

  .SelectedStage {
    align-items: flex-start;
    display: flex;
    padding: 0 10px;
    vertical-align: top;

    @include breakpoint($md) {
      align-items: center;
      vertical-align: middle;  
    }

    .stage {
      display: block;
      position: sticky;
      top: 0;
      width: 100%;
    }
    
    .stageCard {
      background-color: black;
      border: 6px solid black;
      border-top-width: 12px;
      clip-path: polygon(
        0% 9%,     /* top left */
        7% 0%,     /* top left */
        100% 0%,    /* top right */
        100% 0%,   /* top right */
        100% 0%,  /* bottom right */
        100% 100%,  /* bottom right */
        0% 100%,   /* bottom left */
        0 100%      /* bottom left */
      );
      max-width: 100%;

      img {
        clip-path: polygon(
          0% 13%,     /* top left */
          9% 0%,     /* top left */
          100% 0%,    /* top right */
          100% 0%,   /* top right */
          100% 0%,  /* bottom right */
          100% 100%,  /* bottom right */
          0% 100%,   /* bottom left */
          0 100%      /* bottom left */
        );
        border: 0;
        // border: 6px solid white;
        width: auto;
      }

      .name {
        font-size: 24px;
        line-height: 1em;
      }

      .namePlate {
        background-image: radial-gradient(ellipse, #546171 50%, #37405A 50%);
        background-position: 20% 32%;
        background-size: 150% 1500%;
        border: none;
        height: auto;
        padding: 0.5em 2px;
      }

      .selectedStageToken {
        display: none;
        visibility: hidden;
      }
    }
  }

  .StagePicker{
    display: grid;
    grid-template-columns: repeat(2, auto);
    margin-bottom: auto;
    overflow-y: scroll;

    @include breakpoint($sm) {
      grid-template-columns: repeat(3, auto);
    }

    @include breakpoint($md) {
      grid-template-columns: repeat(5, auto);
    }

    @include breakpoint($lg) {
      grid-template-columns: repeat(6, auto);
    }

    @include breakpoint($xl) {
      grid-template-columns: repeat(8, auto);
    }

    @include breakpoint($hd) {
      grid-template-columns: repeat(9, auto);
    }
  }
}
