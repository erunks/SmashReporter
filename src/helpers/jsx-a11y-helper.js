export function buttonize(action) {
  return {
    role: 'button',
    onClick: action,
    onKeyDown: (event) => {
      // Handles Enter/Return pressed
      if (event.keycode === 13) {
        action(event);
      }
    }
  };
}

export default { buttonize };
