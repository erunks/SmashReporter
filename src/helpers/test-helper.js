import React from 'react';
import { render } from '@testing-library/react';
import { mount, shallow } from 'enzyme';
import { IKContext } from 'imagekitio-react';
import { RecoilRoot } from 'recoil';

const wrapper = (component, options = {
  initializeState: () => {},
}) => {
  const initialStateMethod = options.initializeState;

  return (
    <IKContext
      publicKey="public_DfZPshOEvWdxzvxS8HGNWPJCo70="
      urlEndpoint="https://ik.imagekit.io/ozcerk4wii"
    >
      <RecoilRoot initializeState={ initialStateMethod }>
        { component }
      </RecoilRoot>
    </IKContext>
  );
};

const mountWrapper = (component, options) => mount(wrapper(component, options));
const shallowWrapper = (component, options) => shallow(wrapper(component, options));
const wrappedRender = (component, options = {}) => {
  const initialStateMethod = options && options.initializeState;

  const Wrapper = ({ children }) => (
    <IKContext
      publicKey="public_DfZPshOEvWdxzvxS8HGNWPJCo70="
      urlEndpoint="https://ik.imagekit.io/ozcerk4wii"
    >
      <RecoilRoot initializeState={ initialStateMethod }>
        { children }
      </RecoilRoot>
    </IKContext>
  );

  return render(component, { wrapper: Wrapper, ...options });
};


export * from '@testing-library/react';

export {
  mountWrapper,
  shallowWrapper,
  wrappedRender
};
