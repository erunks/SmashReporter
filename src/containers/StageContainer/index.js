import React, { useEffect, useState } from 'react';
import propTypes from 'prop-types';
import { getStages, getLegalStages } from '../../lib/api';
import Stage from '../../components/Stage';

import './StageContainer.scss';

const StageContainer = ({ legal = true, initStages = [{}] }) => {
  const [stages, setStages] = useState(initStages);

  useEffect(() => {
    if (legal) {
      getLegalStages().then((legalStages) => {
        setStages(legalStages);
      });
    } else {
      getStages().then((stages) => {
        setStages(stages);
      });
    }
  }, [legal]);

  const stagesPopulated = () => Object.getOwnPropertyNames(stages[0]).length

  return (
    <div className="StageContainer">
      {stagesPopulated() && stages.map((stageData) => <Stage key={stageData.id} {...stageData} />)}
    </div>
  );
};

const {
  arrayOf,
  bool,
  object
} = propTypes;

StageContainer.propTypes = {
  legal: bool,
  initStages: arrayOf(object)
};

export default StageContainer;
