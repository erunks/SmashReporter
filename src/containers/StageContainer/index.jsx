import React from 'react';
import propTypes from 'prop-types';
import withStageData from '../../hoc/withStageData/index.jsx';
import Stage from '../../components/Stage/index.jsx';

import './StageContainer.scss';

const StageContainer = ({ legal, stages = [] }) => {
  const stageDataPopulated = () => Object.getOwnPropertyNames(stages[0]).length

  return (
    <div className="StageContainer">
      {stageDataPopulated() && stages.map((stageData) => <Stage key={stageData.id} {...stageData} />)}
    </div>
  );
};

const {
  arrayOf,
  bool,
  object
} = propTypes;

StageContainer.propTypes = {
  legal: bool.isRequired,
  stages: arrayOf(object).isRequired
};

export default withStageData(StageContainer);
