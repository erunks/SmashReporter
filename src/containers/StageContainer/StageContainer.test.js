import React from 'react';
import { mountWrapper } from 'helpers/test-helper';
import StageContainer from '.';

describe('<StageContainer>', () => {
  const props = {
    legal: true,
    initStages: [
      {
        dlc: 0,
        id: 1,
        name: 'Battlefield'
      },
      {
        dlc: 0,
        id: 3,
        name: 'Final Destination'
      }
    ]
  };

  const container = mountWrapper(<StageContainer {...props} />);

  it('should match the snapshot', () => {
    expect(container.html()).toMatchSnapshot();
  });

  it('should render the selected stage', () => {
    expect(container.find('.SelectedStage > Stage').length).toEqual(1);
  });

  it('should render the same number of Stage components as stages', () => {
    expect(container.find('.StagePicker > Stage').length).toEqual(props.initStages.length);
  });
});
