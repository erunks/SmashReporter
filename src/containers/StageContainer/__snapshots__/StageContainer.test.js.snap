// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<StageContainer> should match the snapshot 1`] = `"<div class=\\"MainLayout\\"><header class=\\"Header\\"><h1>Smash Reporter</h1></header><main><div class=\\"StageContainer\\"><div class=\\"stage\\"><div class=\\"stageCard\\"><img src=\\"https://www.smashbros.com/assets_v2/img/stage/stage_img1.jpg\\" alt=\\"stage: Battlefield\\"/><div class=\\"namePlate\\"><div class=\\"name\\">Battlefield</div></div></div></div><div class=\\"stage\\"><div class=\\"stageCard\\"><img src=\\"https://www.smashbros.com/assets_v2/img/stage/stage_img3.jpg\\" alt=\\"stage: Final Destination\\"/><div class=\\"namePlate\\"><div class=\\"name\\">Final Destination</div></div></div></div></div></main><footer class=\\"Footer\\"><span>Edward Runkel<i>Â©2020</i></span></footer></div>"`;
